### PR TITLE
Cross-chain QRC20X support and bugfixes

### PR DESCRIPTION
**Pull request by jdowning100**

### CODE REVIEW

File: background/redux-slices/assets.ts [hunk 1]
```diff
@@ -258,13 +258,30 @@ export const transferAsset = createBackgroundAsyncThunk(
...
-        let transactionDetails = await token.populateTransaction.transfer(toAddress, assetAmount.amount)
-        toAddress = transactionDetails.to? transactionDetails.to : ""
-        data = transactionDetails.data? transactionDetails.data : ""
-        assetAmount = {
-          asset: assetAmount.asset,
-          amount: BigInt(0),
+        let fromShard = getShardFromAddress(fromAddress)
+        let toShard = getShardFromAddress(toAddress)
+        if(fromShard != toShard) { // ERC20x cross-chain transfer
+          let minerTip = BigInt(1000000000) * BigInt(calcEtxFeeMultiplier(fromShard, toShard))
```
Comment (wizeguyy) [inline background/redux-slices/assets.ts:261]: Is it possible to not have hard coded magic numbers here?

File: background/services/signing/index.ts [hunk 2]
```diff
@@ -185,7 +185,8 @@ export default class SigningService extends BaseService<Events> {
     // Refresh connected Ledger info indiscriminately. No real harm vs doing it
     // only under certain circumstances, might even be more performant than
     // testing whether the Ledger can sign for the requested `from` address.
-    await this.ledgerService.refreshConnectedLedger()
+    // There is no longer ledger support.
+    // await this.ledgerService.refreshConnectedLedger()
```
Comment (gameofpointers) [inline background/services/signing/index.ts]: comment?

**Review by wizeguyy — APPROVED** [review]

File: background/redux-slices/assets.ts (hunk 1 above)
Comment (jdowning100) [reply background/redux-slices/assets.ts:261]: https://github.com/PelagusWallet/pelagus-extension/issues/7
Still an open issue

File: background/main.ts [hunk 3]
```diff
@@ -545,13 +545,19 @@ export default class Main extends BaseService<never> {
...
+          }
           newBalance = (await this.chainService.assetData.getTokenBalance(selectedAccount, asset.contractAddress)).amount
         } else if (isBuiltInNetworkBaseAsset(asset, selectedAccount.network)) {
           newBalance = (await this.chainService.getLatestBaseAccountBalance(selectedAccount)).assetAmount.amount
         } else {
           logger.error("Unknown asset type for balance checker, asset: " + asset.symbol)
-          return
+          continue
         }
+        isSmartContractFungibleAsset(asset) ? console.log("Balance checker: " + asset.symbol + " " + newBalance + " " + asset.contractAddress)
```
Comment (gameofpointers) [inline background/main.ts:559]: is this log required ?

File: background/redux-slices/assets.ts (hunk 1 above)
Comment (gameofpointers) [reply background/redux-slices/assets.ts:261]: Gwei

File: background/redux-slices/assets.ts [hunk 4]
```diff
@@ -258,13 +258,30 @@ export const transferAsset = createBackgroundAsyncThunk(
...
+            asset: QUAI,
+            amount: amount,
+          }
+        } else {
+          let transactionDetails = await token.populateTransaction.transfer(toAddress, assetAmount.amount)
+          toAddress = transactionDetails.to? transactionDetails.to : ""
+          data = transactionDetails.data? transactionDetails.data : ""
+          assetAmount = {
+            asset: QUAI,
+            amount: BigInt(0),
```
Comment (gameofpointers) [inline background/redux-slices/assets.ts:278]: why is the amount zero

Comment (jdowning100) [reply background/redux-slices/assets.ts:278]: In a regular intra-shard ERC20 transfer, the 'amount' field is zero.

File: background/main.ts (hunk 3 above)
Comment (jdowning100) [reply background/main.ts:559]: Yes, it helps me with debugging

**Review by gameofpointers — DISMISSED** [review]

**Review by wizeguyy — DISMISSED** [review]

File: background/services/signing/index.ts (hunk 2 above)
Comment (jdowning100) [reply background/services/signing/index.ts]: Fixed

**Review by gameofpointers — APPROVED** [review]

**Review by wizeguyy — APPROVED** [review]

**Pull request merged**